include/boost/geometry/util/math.hpp

```diff
@@ -191,6 +192,66 @@ struct square_root<T, true>
 };
 
 
+
+template
+<
+    typename T,
+    bool IsFundemantal = boost::is_fundamental<T>::value /* false */
+>
+struct modulo
+{
+    typedef T return_type;
+
+    static inline T apply(T const& value1, T const& value2)
+    {
+        // for non-fundamental number types assume that sqrt is
+        // defined either:
+        // 1) at T's scope, or
+        // 2) at global scope, or
+        // 3) in namespace std
+        using ::fmod;
+        using std::fmod;
+
+        return fmod(value1, value2);
+    }
+};
+
+template <typename Fundamental, bool IsIntegral>
```

Comment:
True: `IsIntergral` can be initialized as you propose. Will do it your way.

All three versions you mention are already implemented, the difference being that for user defined functions I assume the presence of `fmod()` rather than `mod()`.
I agree that `mod()` is a more generic name. The only instance of user-defined number type that I have and have used is `ttmath::Big<>`, and for those I have added myself the missing function (see PR #268). So we are pretty flexible here. Using `mod()` as the required function is okay with me and will do the change.